lib/translator/read_capacity_10.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "read_capacity_10.h"
+
+namespace translator {
+
+StatusCode ReadCapacity10ToNvme(absl::Span<const uint8_t> raw_scsi,
+                                nvme::GenericQueueEntryCmd& identify_ns,
+                                scsi::LunAddress lun, Allocation allocation) {
+  scsi::ReadCapacity10Command cmd = {};
+  if (!ReadValue(raw_scsi, cmd)) {
+    DebugLog("Malformed ReadCapacity10 Command - Error in reading to buffer");
+    return StatusCode::kInvalidInput;
+  };
+
+  if (cmd.control_byte.naca == 1) {
+    DebugLog("Malformed ReadCapacity10 Command - Invalid NACA bit");
+    return StatusCode::kInvalidInput;
+  }
+
+  // Allocate prp & assign to command
+  auto read_status = allocation.SetPages(1, 0);
+  if (read_status != StatusCode::kSuccess) {
+    return read_status;
+  }
+
+  identify_ns = nvme::GenericQueueEntryCmd{
+      .opc = static_cast<uint8_t>(nvme::AdminOpcode::kIdentify), .nsid = lun};
+  identify_ns.dptr.prp.prp1 = allocation.data_addr;
+  identify_ns.cdw[0] = 0x0;  // cns val
+
+  return StatusCode::kSuccess;
+}
+
+StatusCode ReadCapacity10ToScsi(absl::Span<uint8_t> buffer,
+                                nvme::GenericQueueEntryCmd gen_identify_ns) {
```

Comment:
Why do we need an entire GenericQueueEntryCmd as input? Is this a convention we are using for all `ToScsi` functions? Would it be cleaner to pass as an input the value of `gen_identify_ns.dptr.prp.prp1` instead?

Reply:
this is currently a convention for our generic translator. we can create a separate issue to change this standard if needed